lib/scope.js

```diff
@@ -746,34 +745,29 @@ AST_Toplevel.DEFMETHOD("compute_char_frequency", function(options) {
     }
 });
 
-var base54 = (function() {
-    var leading = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ$_".split("");
-    var digits = "0123456789".split("");
-    var chars, frequency;
-    function reset() {
-        frequency = new Map();
-        leading.forEach(function(ch) {
-            frequency.set(ch, 0);
-        });
-        digits.forEach(function(ch) {
-            frequency.set(ch, 0);
-        });
-    }
-    base54.consider = function(str, delta) {
-        for (var i = str.length; --i >= 0;) {
+const base54 = (function() {
+    const freq = new Map();
+
+    const digits = init("0123456789");
+    const leading = init("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ$_");
```

Comment:
`sort` method only used for two arrays above which have 10 and 54 elements

Reply:
It looks like the [threshold between insert sort (stable) and quick sort (unstable) is 10 elements](https://github.com/v8/v8/blob/6.9.427.17/src/js/array.js#L647). So `digits` would always use insertion sort, but `leading` wouldn't. 😕